jetstream/cli.py

```diff
@@ -42,6 +43,124 @@ def setup_logger(
 logger = logging.getLogger(__name__)
 
 
+RECOGNIZED_EXPERIMENT_TYPES = ("pref", "addon", "message", "v6")
+
+
+@attr.s
+class AllType:
+    """Sentinel value for AnalysisExecutor"""
+
+
+All = AllType()
+
+
+class ExecutorStrategy(Protocol):
+    project_id: str
+    dataset_id: str
+
+    def __init__(self, project_id: str, dataset_id: str, *args, **kwargs) -> None:
+        ...
+
+    def execute(self, worklist: Iterable[Tuple[str, AnalysisSpec, datetime]]) -> bool:
+        ...
+
+
+@attr.s(auto_attribs=True)
+class SerialExecutorStrategy:
+    project_id: str
+    dataset_id: str
+    analysis_class: Type = Analysis
+    experiment_getter: Callable[[], ExperimentCollection] = ExperimentCollection.from_experimenter
+
+    def execute(self, worklist):
+        failed = False
+        experiments = self.experiment_getter()
+        for slug, spec, date in worklist:
+            try:
+                experiment = experiments.with_slug(slug).experiments[0]
+                config = spec.resolve(experiment)
+                self.analysis_class(self.project_id, self.dataset_id, config).run(date)
+            except Exception as e:
+                failed = True
+                logger.exception(str(e), exc_info=e, extra={"experiment": slug})
+        return not failed
```

Comment:
I'm not sure if having a `ExecutorStrategy` here is necessary. I started incorporating the new CLI in https://github.com/mozilla/jetstream/pull/301 and made some [changes to `AnalysisExecutor` that remove `SerialExecutorStrategy`](https://github.com/mozilla/jetstream/pull/301/files#r515409137)

Reply:
`ExecutorStrategy` seems like a good option here. We can use it to implement an `ArgoExecutionStrategy` which will pass all active experiments as parameters into the Argo workflow yaml configuration. This way we wouldn't need a `get_active_experiments` entrypoint